src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -255,7 +253,7 @@ public File asFile() {
 
     @Override
     public Path asPath() {
-        return fileResource.toPath();
+        return IOUtil.toPath(fileResource);
```

Comment:
This code makes unchecked references to `fileResource` elsewhere; why is it expected that it's OK for it to be null here?